pkg/openstack/common.go

```diff
@@ -410,13 +410,18 @@ func (ed *EndpointDetail) CreateRoute(
 	owner metav1.Object,
 ) (ctrl.Result, error) {
 	// initialize the route with any custom provided route override
+	targetPortName := ed.Service.Spec.ObjectMeta.Annotations[service.AnnotationIngressTargetPortNameKey]
+	if targetPortName == "" {
```

Comment:
should it be like this to check if the annotation exists and has a value?
```
// per default use the service name as targetPortName if we don't have the annotation.
targetPortName = ed.Service.Spec.Name
if name, ok := ed.Service.Spec.ObjectMeta.Annotations[service.AnnotationIngressTargetPortNameKey]; ok && name != ""{
        targetPortName = name
}
```

Reply:
When I was testing this, I tested with a service which had the annotation and with a service, which didn't have it. Both cases behaved correctly. If I understand the suggestion correctly, it should also deal with the case, when the annotation isn't defined in lib-common, which was the issue of this PR before updating it. I feel like it's also a little bit more visible what's the default behavior. I'll try it out and use it if it works :+1:

Reply:
it would also deal with the situation where the annotation is defined, but the value is empty. so the annotation would only be used if it is set and has a non empty value

Reply:
So it didn't help with the issue, when the annotation isn't defined in lib-common. Guess that's resolved during compile time (makes sense). As far as I can tell the functionality is the same, but I'll leave it in. Thanks for the suggestion.